src/image.c

```diff
@@ -82,12 +144,25 @@ void image_thumbnail(struct image* image, size_t size, bool fill,
         offset_y = 0;
     }
 
+    // get thumbnail from cache
+    if (get_image_thumb_path(image, thumb_path) &&
```

Comment:
`image_thumbnail` is intended to create a thumbnail, not load.
Any function should do one thing =)

Reply:
Should I change gallery.c: add_thumbnail() to first try to get thumbnail from cache and if it is unavailable, then run image_thumbnail() and store it on disk?